### PR TITLE
Power Checker Updates (Vents!)

### PR DESCRIPTION
**Pull request by Azrun**
[feature]
## About the PR 
Added vent capture machinery to Power Checker.
Rebuild `power/collector/control/collector/controlers` list on refresh to account for them being constructable.
Added check for z level.

Resolves runtime that occurs with "golden" TEG variant where the generator is exploded/deleted.

## Why's this needed? 
Improves information provided to engineering and others regarding power sources on Oshan and Singularity maps.

```
(u)Azrun:
(+)Added vent capture units to Power Checker 0.14! Rebuild lists of constructable power devices.
```

### CODE REVIEW

File: code/obj/item/device/pda2/smallprogs.dm [hunk 1]
```diff
@@ -514,79 +514,106 @@ Code:
...
+		if(!ref || ref.disposed)
+			ref = locate(type) in machine_registry[MACHINES_POWER]
+			if(ref?.z != 1) ref = null
+		. = ref
+
+	proc/find_machinery_list(list/obj_list, type)
+		obj_list.len = 0
+		for(var/obj/M in machine_registry[MACHINES_POWER])
+			if(istype(M, type) && M.z == 1)
+				obj_list.Add(M)
```
Comment (ZeWaka) [inline code/obj/item/device/pda2/smallprogs.dm:535]: ..................why are you not just using by_type

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/item/device/pda2/smallprogs.dm (hunk 1 above)
Comment (Azrun) [reply code/obj/item/device/pda2/smallprogs.dm:535]: Excuse: O(n) vs O(n)
Excuse: Space savings, I fall back mentally to embedded space constraints.
Excuse: List of things currently using those lists seems so small, so adding objects that there are typically 0 to 5 of seemed odd.

Will update, good idea. 👍

Comment (ZeWaka) [reply code/obj/item/device/pda2/smallprogs.dm:535]: not O(n) vs O(N) btw, O(1) vs O(N)

Comment (Azrun) [reply code/obj/item/device/pda2/smallprogs.dm:535]: 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦
I thought you were just talking about line 533, and I was thinking I still wanted to filter on z level.  My bad.  When you're right, you're right. 😉 

No excuses.

File: code/obj/item/device/pda2/smallprogs.dm [hunk 2]
```diff
@@ -513,80 +513,85 @@ Code:
...
-
-			stuff += "Power Stored: [engineering_notation(laser.charge)]J ([round(100.0*laser.charge/laser.capacity, 0.1)]%)<BR>"
-			stuff += "Power Input: [engineering_notation(laser.chargelevel)]W<BR>"
-			stuff += "Power Output: [engineering_notation(laser.output)]W<BR>"
+					. += "<BR><h4>Radiation Collector [controler_index++] Status</h4>"
+					. += "Output: [engineering_notation(C.lastpower)]W<BR>"
+					if(C.CA1?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P1 ? round(MIXTURE_PRESSURE(C.P1?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA2?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P2 ? round(MIXTURE_PRESSURE(C.P2?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA3?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P3 ? round(MIXTURE_PRESSURE(C.P3?.air_contents), 0.1) : "ERR"] kPa<BR>"
+					if(C.CA4?.active) . += "Collector [collector_index++]: Tank Pressure: [C.P4 ? round(MIXTURE_PRESSURE(C.P4?.air_contents), 0.1) : "ERR"] kPa<BR>"
```
Comment (ZeWaka) [inline code/obj/item/device/pda2/smallprogs.dm:572]: you don't need the inner ?.s on the PX vars

Comment (Azrun) [reply code/obj/item/device/pda2/smallprogs.dm]: Removed redundancy.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**